Neos.ContentRepository.Core/Classes/NodeType/ConstraintCheck.php

```diff
@@ -0,0 +1,137 @@
+<?php
+namespace Neos\ContentRepository\Core\NodeType;
+
+/**
+ * Performs node type constraint checks against a given set of constraints
+ * @internal
+ */
+class ConstraintCheck
+{
+    /**
+     * @param array<string,mixed> $constraints
+     */
+    public function __construct(
+        private readonly array $constraints
```

Comment:
Would be neat to have this be a DTO with the constraints configuration.

Reply:
yes or at lest an example how this looks would be cool ^^